processor/lsmintervalprocessor/processor.go

```diff
@@ -359,16 +359,59 @@ func (p *Processor) exportForInterval(
 
 	var errs []error
 	var exportedDPCount int
-	ivlStr := formatDuration(ivl)
 	for iter.First(); iter.Valid(); iter.Next() {
 		var v merger.Value
 		if err := v.UnmarshalProto(iter.Value()); err != nil {
 			errs = append(errs, fmt.Errorf("failed to decode binary from database: %w", err))
 			continue
 		}
-		for i := 0; i < v.Metrics.ResourceMetrics().Len(); i++ {
-			res := v.Metrics.ResourceMetrics().At(i)
-			res.Resource().Attributes().PutStr("metricset.interval", ivlStr)
+		resourceMetrics := v.Metrics.ResourceMetrics()
+		if ivl.Statements != nil {
+			for i := 0; i < resourceMetrics.Len(); i++ {
+				res := resourceMetrics.At(i)
+				scopeMetrics := res.ScopeMetrics()
+				for j := 0; j < scopeMetrics.Len(); j++ {
+					scope := scopeMetrics.At(j)
+					metrics := scope.Metrics()
+					for k := 0; k < metrics.Len(); k++ {
+						metric := metrics.At(k)
+						executeTransform := func(dp any) {
+							dCtx := ottldatapoint.NewTransformContext(dp, metric, metrics, scope.Scope(), res.Resource(), scope, res)
+							if err := ivl.Statements.Execute(ctx, dCtx); err != nil {
+								errs = append(errs, fmt.Errorf("failed to execute ottl statement for interval %s: %w", ivl.Duration, err))
+							}
+						}
+						// TODO (lahsivjar): add exhaustive:enforce lint rule
```

Comment:
[**For reviewers**] Will create follow-up PR for this: https://github.com/elastic/opentelemetry-collector-components/issues/137